# 7-base_geometry.py

This module contains a class base_geometry

Importing:
>>> BaseGeometry = __import__('7-base_geometry').BaseGeometry

Tests:
>>> bg = BaseGeometry()
>>> bg.integer_validator("my_int", 12)
>>> bg.integer_validator("width", 89)
>>> bg.integer_validator("height", 1)
>>> bg.integer_validator("name", "John")
Traceback (most recent call last):
	...
TypeError: name must be an integer
>>> bg.integer_validator("age", 0)
Traceback (most recent call last):
	...
ValueError: age must be greater than 0
>>> bg.integer_validator("distance", -4)
Traceback (most recent call last):
	...
ValueError: distance must be greater than 0
>>> bg.integer_validator("distance", 0)
Traceback (most recent call last):
	...
ValueError: distance must be greater than 0
>>> bg.integer_validator("distance", 4.5)
Traceback (most recent call last):
	...
TypeError: distance must be an integer
>>> bg.integer_validator("distance", [4, 5])
Traceback (most recent call last):
	...
TypeError: distance must be an integer
>>> bg.integer_validator("distance", (4, 5))
Traceback (most recent call last):
	...
TypeError: distance must be an integer
>>> bg.integer_validator("distance", {4, 5})
Traceback (most recent call last):
	...
TypeError: distance must be an integer
>>> bg.integer_validator("distance", {4: 5})
Traceback (most recent call last):
	...
TypeError: distance must be an integer
>>> bg.integer_validator("distance", True)
Traceback (most recent call last):
	...
TypeError: distance must be an integer
>>> bg.integer_validator("distance", None)
Traceback (most recent call last):
	...
TypeError: distance must be an integer
>>> bg.area()
Traceback (most recent call last):
	...
Exception: area() is not implemented
